modules/kafka/kafka_test.go

```diff
@@ -97,6 +100,140 @@ func TestKafka_invalidVersion(t *testing.T) {
 	}
 }
 
+func TestKafka_network(t *testing.T) {
+	ctx := context.Background()
+	var err error
+
+	const (
+		topic_in  = "topic_in"
+		topic_out = "topic_out"
+	)
+
+	Network, err := network.New(ctx, network.WithCheckDuplicate())
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// kafkaWithListener {
+	KafkaContainer, err := kafka.RunContainer(ctx,
+		kafka.WithClusterID("test-cluster"),
+		testcontainers.WithImage("confluentinc/confluent-local:7.6.1"),
+		network.WithNetwork([]string{"kafka"}, Network),
+		kafka.WithListener([]kafka.KafkaListener{
+			{
+				Name: "INTERNAL",
+				Ip:   "kafka",
+				Port: "9092",
+			},
+		}),
+	)
+	// }
+
+	if err != nil {
+		t.Fatalf("failed to start container: %s", err)
+	}
+
+	brokers, err := KafkaContainer.Brokers(context.TODO())
+	if err != nil {
+		t.Fatal("failed to get brokers", err)
+	}
+
+	createTopics(brokers, []string{topic_in, topic_out})
+
+	initKafkaTest(ctx, Network.Name, "kafka:9092", topic_in, topic_out)
+
+	// perform assertions
+
+	// set config to true because successfully delivered messages will be returned on the Successes channel
+	config := sarama.NewConfig()
+	config.Producer.Return.Successes = true
+
+	producer, err := sarama.NewSyncProducer(brokers, config)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Act
+	key := "wow"
+	text_msg := "test-input-external"
+
+	if _, _, err := producer.SendMessage(&sarama.ProducerMessage{
+		Topic: topic_in,
+		Key:   sarama.StringEncoder(key),
+		Value: sarama.StringEncoder(text_msg),
+	}); err != nil {
+		t.Fatal(err)
+	}
+
+	client, err := sarama.NewConsumerGroup(brokers, "groupName", config)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	consumer, _, done, cancel := NewTestKafkaConsumer(t)
+	go func() {
+		if err := client.Consume(context.Background(), []string{topic_out}, consumer); err != nil {
+			cancel()
+		}
+	}()
+
+	// wait for the consumer to be ready
+	<-done
+
+	if consumer.message == nil {
+		t.Fatal("Empty message")
+	}
+
+	// Assert
+	if !strings.Contains(string(consumer.message.Value), text_msg) {
+		t.Error("got wrong string")
+	}
+}
+
+func initKafkaTest(ctx context.Context, network string, brokers string, input string, output string) (testcontainers.Container, error) {
+	req := testcontainers.ContainerRequest{
+		FromDockerfile: testcontainers.FromDockerfile{
+			Context:       "./docker",
```

Comment:
Can we rename the `docker` directory to `testdata`? Just for consistency and because the toolchain with consider that dir as part of the tests, not a real module.

Reply:
Done